rspec-tools/rspec_tools/validation/description.py

```diff
@@ -1,55 +1,159 @@
 from pathlib import Path
 from typing import Final
+from collections import Counter
 
 from rspec_tools.errors import RuleValidationError
 from rspec_tools.rules import LanguageSpecificRule
 from rspec_tools.utils import LANG_TO_SOURCE
 
 import re
 
+def read_file(path):
+  section_names_path = Path(__file__).parent.parent.parent.parent.joinpath(path)
+  return section_names_path.read_text(encoding='utf-8').split('\n')
+
+def parse_names(path):
+  section_names_path = read_file(path)
+  return [s.replace('* ', '').strip() for s in section_names_path if s.strip()]
+
+HOW_TO_FIX_IT = 'How to fix it'
+HOW_TO_FIX_IT_REGEX = re.compile(HOW_TO_FIX_IT)
+
+def parse_education_section_names(path):
+  education_format_file = read_file(path)
+  sections = {}
+  optional_sections = {}
+  current_map = sections
+  current_section_name = ''
+  for line in education_format_file:
+    if line.startswith('== '):
+      section = line.replace('== ', '').strip()
+      if section.endswith('(optional)'):
+        section = section.replace(' (optional)', '')
+        current_map = optional_sections
+      else:
+        current_map = sections
+      if HOW_TO_FIX_IT_REGEX.match(section):
+        current_map[HOW_TO_FIX_IT] = set()
+        current_section_name = HOW_TO_FIX_IT
+      else:
+        current_section_name = section
+        current_map[section] = set()
+    if line.startswith('=== '):
+      section = line.replace('=== ', '').strip()
+      current_map[current_section_name].add(section)
+  return [
+    sections,
+    optional_sections,
+  ]
+
 # The list of all the sections currently accepted by the script.
 # The list includes multiple variants for each title because they all occur
 # in the migrated RSPECs.
 # Further work required to shorten the list by renaming the sections in some RSPECS
 # to keep only on version for each title.
-SECTION_NAMES_PATH = Path(__file__).parent.parent.parent.parent.joinpath('docs/section_names.adoc')
-SECTION_NAMES_FILE = SECTION_NAMES_PATH.read_text(encoding='utf-8').split('\n')
-ACCEPTED_SECTION_NAMES: Final[list[str]] = [s.replace('* ', '').strip() for s in SECTION_NAMES_FILE if s.strip()]
+LEGACY_SECTION_NAMES: Final[list[str]] = parse_names('docs/header_names/legacy_section_names.adoc')
 # The list of all the framework names currently accepted by the script.
-FRAMEWORK_NAMES_PATH = Path(__file__).parent.parent.parent.parent.joinpath('docs/allowed_framework_names.adoc')
-FRAMEWORK_NAMES_FILE = FRAMEWORK_NAMES_PATH.read_text(encoding='utf-8').split('\n')
-ACCEPTED_FRAMEWORK_NAMES: Final[list[str]] = [s.replace('* ', '').strip() for s in FRAMEWORK_NAMES_FILE if s.strip()]
+ACCEPTED_FRAMEWORK_NAMES: Final[list[str]] = parse_names('docs/header_names/allowed_framework_names.adoc')
+
+[
+  SECTIONS,
+  OPTIONAL_SECTIONS
+  ] = parse_education_section_names('docs/header_names/education_format_names.adoc')
+
+
+
+def intersection(lst1, lst2):
+  lst3 = [value for value in lst1 if value in lst2]
+  return lst3
```

Comment:
```suggestion
  return list(set(lst1).intersection(lst2))
```
In harmony with the `difference` function you have below